packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -93,33 +1691,224 @@ abstract class TreeDiagnosticsMixin {
+  ///    enum types.
+  ///  * [IterableProperty], which handles iterable values with display
+  ///    customizable depending on the [DiagnosticsTreeStyle] used.
+  ///  * [LazyDiagnosticsProperty], which handles properties where computing the
+  ///    value could throw an exception.
+  ///  * [ObjectFlagProperty], which provides terse descriptions of whether a
+  ///    property value is present or not. For example, whether an `onClick`
+  ///    callback is specified or an animation is in progress.
+  ///
+  /// If non of these subclasses apply, use the [DiagnosticProperty]
+  /// constructor or in rare cases create your own [DiagnosticsProperty]
+  /// subclass as in the case for [TransformProperty] which handles [Matrix4]
+  /// that represent transforms. Generally any property value with a good
+  /// toString() implementation works fine using [DiagnosticProperty] directly.
+  ///
+  /// ## Sample code
+  ///
+  /// This example shows best practices for implementing debugFillProperties
+  /// illustrating use of all common [DiagnosticProperty] subclasses and all
+  /// common [DiagnosticsProperty] parameters.
+  ///
+  /// ```dart
+  /// @override
+  /// void debugFillProperties(List<DiagnosticsNode> properties) {
+  ///   // Always add properties from the base class first.
+  ///   super.debugFillProperties(properties);
+  ///
+  ///   // Omit the property name 'message' when displaying this String property
+  ///   // as it would just add visual noise.
+  ///   description.add(new StringProperty('message', message, quoted: true, showName: false));
+  ///
+  ///   description.add(new DoubleProperty('stepWidth', stepWidth));
+  ///
+  ///   // A scale of 1.0 does nothing so should be hidden.
+  ///   description.add(new DoubleProperty('scale', scale, defaultValue: 1.0));
+  ///
+  ///   // If the hitTestExtent matches the paintExtent, it is just set to its
+  ///   // default value so is not relevant.
+  ///   properties.add(new DoubleProperty('hitTestExtent', hitTestExtent, defaultValue: paintExtent));
+  ///
+  ///   // maxWidth of double.INFINITY indicates the width is unconstrained and
+  ///   // so maxWidth has no impact.,
+  ///   description.add(new DoubleProperty('maxWidth', maxWidth, defaultValue: double.INFINITY));
+  ///
+  ///   // Progress is a value between 0 and 1 or null. Showing it as a
+  ///   // percentage makes the meaning clear enough that the name can be
+  ///   // hidden.
+  ///   description.add(new PercentProperty(
+  ///     'progress',
+  ///     progress,
+  ///     showName: false,
+  ///     nullDescription: '<indeterminate>',
+  ///   ));
+  ///
+  ///   // Most text fields have maxLines set to 1.
+  ///   description.add(new IntProperty('maxLines', maxLines, defaultValue: 1));
+  ///
+  ///   // Specify the unit as otherwise it would be unclear that time is in
+  ///   // milliseconds.
+  ///   description.add(new IntProperty('duration', duration.inMilliseconds, unit: 'ms'));
```

Comment:
What about making the toString for duration smarter so you can just write
new DiagnosticsProperty<Duration>('duration', duration)
I don't have background on why duration is shown as 25:33:00.000500
At minimum I would expect it could be helpful to omit hours and minutes when they are zero
and showing 6 significant digits for seconds seems like overkill.
For example, the toString could take an optional precision parameter which could default to
displaying the value in ms.